submissions/karelskiy/tiny-JS-world/index.js

```diff
@@ -0,0 +1,52 @@
+const dog = {
+    species: 'dog',
+    name: 'Toby',
+    gender: 'male',
+    legs: 4,
+    hands: 0,
+    saying: 'woof-woof!',
+};
+const cat = {
+    species: 'cat',
+    name: 'Mrs. Velasquez',
+    gender: 'female',
+    love: 'eat',    
+    legs: 4,
+    hands: 0,
+    saying: 'meow-meow!'
+};
+const Enrico = {
+    species: 'human',
+    name: 'Enrico',
+    gender: 'male',
+    legs: 2,
+    hands: 2,
+    saying: 'Holla'
+};
+const Angel = {
+    species: 'human',
+    name: 'Angel',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    saying: 'Ehi, buongiorno'
+};
+
+const out = obj => {
+    const MAIN_PROP = ['species', 'name', 'gender', 'legs', 'hands', 'saying'];
+    let result, arr = ['', []];
+    for (let i in obj) {
+        if (MAIN_PROP.indexOf(i) >= 0) {
+            result = `Species: <b>${obj['species']};</b> Name: <b>${obj['name']};</b> Gender: <b>${obj['gender']};</b> Legs: <b>${obj['legs']};</b> Hands: <b>${obj['hands']};</b> Favorite phrase: <b>${obj['saying']};</b> `;
+        } else {
+            arr.push(`<i>${i} <b>${obj[i]};</b></i> `);
+        }
+    }
+    return result + arr.join('');
+}
```

Comment:
The above is a somewhat overly refactoring.
The array was a proper container for objects. Array guarantees order of elements.
All you needed to do to optimize this fragment
```
population.forEach(el => print(Object.values(el).join('; ')));
```
was
```
population.forEach(el => print(['species', 'name', 'gender', 'legs', 'hands', 'saying']
  .map(prop => el[prop])
  .join('; '))
);
```

If some props can be missing by design and you want to take this into account just add an extra check:

```
population.forEach(el => print(['species', 'name', 'gender', 'legs', 'hands', 'saying']
  .filter(prop => el[prop] !== undefined) // test if el has the property defined
  .map(prop => el[prop]) // convert prop names into their values
  .join('; '))
);
```

Reply:
Oh, i finally understand, sorry about my tries.